cpp/src/arrow/flight/sql/types.h

```diff
@@ -70,6 +71,30 @@ struct ARROW_FLIGHT_SQL_EXPORT SqlInfoOptions {
     /// - true: if read only
     FLIGHT_SQL_SERVER_READ_ONLY = 3,
 
+    /// Retrieves a boolean value indicating whether the Flight SQL Server
+    /// supports executing Substrait plans.
+    FLIGHT_SQL_SERVER_SUBSTRAIT = 4,
+
+    /// Retrieves an int32 value indicating whether the Flight SQL Server
+    /// supports explicit transaction RPCs.
+    FLIGHT_SQL_SERVER_TRANSACTION = 5,
```

Comment:
There's already a property SQL_TRANSACTIONS_SUPPORTED, so it might be worth making clear the difference here.

Reply:
Updated the docstring.